### PR TITLE
fea(state): support get_keyed_state function to get data in redis

### PR DESCRIPTION
**Pull request by superrxan**

### CODE REVIEW

File: internal/binder/function/funcs_misc.go [hunk 1]
```diff
@@ -627,6 +628,123 @@ func registerMiscFunc() {
...
+			if !ok {
+				return fmt.Errorf("key %v is not a string", args[0]), false
+			}
+
+			value, err := keyedstate.GetKeyedState(key)
+			if err != nil {
+				return args[2], true
+			}
+
+			if v, ok := args[1].(string); ok {
```
Comment (ngjaying) [inline internal/binder/function/funcs_misc.go:647]: why not use cast package to do the conversion?

File: internal/binder/function/funcs_misc.go [hunk 2]
```diff
@@ -627,6 +628,123 @@ func registerMiscFunc() {
...
+				}
+			} else {
+				return fmt.Errorf("expect string type for the 2nd parameter"), false
+			}
+		},
+		val: func(_ api.FunctionContext, args []ast.Expr) error {
+			if len(args) != 3 {
+				return fmt.Errorf("the args must be three")
+			}
+			if !ast.IsStringArg(args[0]) {
```
Comment (ngjaying) [inline internal/binder/function/funcs_misc.go:738]: This only checks for literal. Refer to other validator functions

File: internal/keyedstate/kv.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package keyedstate
+
+import (
+	"github.com/lf-edge/ekuiper/internal/pkg/store"
+	kv2 "github.com/lf-edge/ekuiper/pkg/kv"
+)
+
+var manager *Manager
+
+type Manager struct {
```
Comment (ngjaying) [inline internal/keyedstate/kv.go:24]: Looks like manager == kv, why split to two levels

File: internal/keyedstate/kv.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	kv kv2.KeyValue
+}
+
+func InitManager() {
+	kv, _ := store.GetKV("keyed_state")
```
Comment (ngjaying) [inline internal/keyedstate/kv.go:29]: This will limit the keys in redis to have a prefix, seems wrong to me

**Review by ngjaying — CHANGES_REQUESTED** [review]
Some problems

File: internal/binder/function/funcs_misc.go (hunk 2 above)
Comment (superrxan) [reply internal/binder/function/funcs_misc.go:738]: ack

File: internal/binder/function/funcs_misc.go (hunk 1 above)
Comment (superrxan) [reply internal/binder/function/funcs_misc.go:647]: ack

**Review by ngjaying — APPROVED** [review]

**Pull request merged**